### ISSUE TITLE
[Bug]: AlertDialogPrepromptForAndroidSettings.show

### ISSUE THREAD

**Issue by Hardik7265** [OneSignal/OneSignal-Unity-SDK#729]
### What happened?

The latest version of SDK cause crash in Samsung Device Galaxy A13.

Error:- Caused by android.view.WindowManager$BadTokenException: Unable to add window -- token android.os.BinderProxy@e4bc233 is not valid; is your activity running?
      

### Steps to reproduce?

```Markdown
1. Install latest SDK.
2. Install build in Samsung Device Galaxy A13.
3. Crash or error in device.
```

### What did you expect to happen?

I expected it should not crash and continue to work.

### Unity version

2023.2.20f1

### OneSignal Unity SDK version

5.1.4

### Platform

Android

### Relevant log output

```Shell
Caused by android.view.WindowManager$BadTokenException: Unable to add window -- token android.os.BinderProxy@e4bc233 is not valid; is your activity running?
       at android.view.ViewRootImpl.setView(ViewRootImpl.java:1764)
       at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:567)
       at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:148)
       at android.app.Dialog.show(Dialog.java:521)
       at android.app.AlertDialog$Builder.show(AlertDialog.java:1153)
       at com.onesignal.core.internal.permissions.AlertDialogPrepromptForAndroidSettings.show(AlertDialogPrepromptForAndroidSettings.kt:65)
       at com.onesignal.notifications.internal.permissions.impl.NotificationPermissionController.showFallbackAlertDialog(NotificationPermissionController.kt:139)
       at com.onesignal.notifications.internal.permissions.impl.NotificationPermissionController.onReject(NotificationPermissionController.kt:124)
       at com.onesignal.core.activities.PermissionsActivity.onRequestPermissionsResult$lambda-0(PermissionsActivity.kt:114)
       at com.onesignal.core.activities.PermissionsActivity.$r8$lambda$_fGz_SOmbIhPQvxRD-41f99MrRo()
       at com.onesignal.core.activities.PermissionsActivity$$ExternalSyntheticLambda0.run(:4)
       at android.os.Handler.handleCallback(Handler.java:942)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loopOnce(Looper.java:226)
       at android.os.Looper.loop(Looper.java:313)
       at android.app.ActivityThread.main(ActivityThread.java:8757)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by shepherd-l** [comment]
Thanks for reporting! We are looking into this

**Comment by shepherd-l** [comment]
Apologies for the late reply, I believe we fixed this in our Unity SDK's [5.1.10 Release](https://github.com/OneSignal/OneSignal-Unity-SDK/releases/tag/5.1.10).

Could you please try updating to the [latest release of our Unity SDK](https://github.com/OneSignal/OneSignal-Unity-SDK/releases/tag/5.1.11)?

If you are still having issues, feel free to comment here and I'll reopen the issue.
